firedrake/slate/slac/compiler.py

```diff
@@ -0,0 +1,324 @@
+"""This is SLATE's Linear Algebra Compiler. This module is responsible for generating
+C++ kernel functions representing symbolic linear algebra expressions written in SLATE.
+
+This linear algebra compiler uses both Firedrake's form compiler, the Two-Stage
+Form Compiler (TSFC) and COFFEE's kernel abstract syntax tree (AST) optimizer. TSFC
+provides this compiler with appropriate kernel functions (in C) for evaluating integral
+expressions (finite element variational forms written in UFL). COFFEE's AST optimizing
+framework produces the resulting kernel AST returned by: `compile_slate_expression`.
+
+The Eigen C++ library (http://eigen.tuxfamily.org/) is required, as all low-level numerical
+linear algebra operations are performed using the `Eigen::Matrix` methods built into Eigen.
+"""
+from __future__ import absolute_import, print_function, division
+
+import petsc
```

Comment:
This is not right if someone built petsc themselves.  Instead do what we do in `setup.py`:

```
def get_petsc_dir():
    try:
        petsc_arch = env.get('PETSC_ARCH', '')
        petsc_dir = env['PETSC_DIR']
        if petsc_arch:
            return (petsc_dir, path.join(petsc_dir, petsc_arch))
        return (petsc_dir,)
    except KeyError:
        try:
            import petsc
            return (petsc.get_petsc_dir(), )
        except ImportError:
            sys.exit("""Error: Could not find PETSc library.  Please report a bug""")
```